File: MantenimientoDaxDirectorio/MantenimientoDirectorio/Directorio.vb
Option Strict On
Option Explicit On

Imports System.Data.SqlClient
Imports DattCom
Public Class Directorio
    ''Esta Clase sirve para cargar directorios en produccion o sistemas que  permito que no usen el adcomdx
    Public codigo As String 'copia local
    Public NombreImpresion As String 'copia local
    Public CiRuc As String 'copia local
    Public telefono1 As String 'copia local
    Public telefono2 As String 'copia local
    Public direccion As String 'copia local
    Public CodVendedor As String 'copia local
    Public ZonaVentas As String 'copia local
    Public TipoCliente As String
    Public Ciudad As String
    Public TipoId As String
    Public NombreCli As String
    Public TipoPersona As String
    'variables locales para almacenar los valores de las propiedades

    Public Function CargarAlex(ByRef CodigoAlex As String, ByRef ClioPro As Boolean, Optional ByRef SoloCodigo As String = "") As Boolean

        Dim apellidos As String = ""
        Dim nombres As String = ""
        Dim seleccion As String = ""

        Dim ConxAdcom As New SqlClient.SqlConnection(datosEmpresa.strConxAdcom)
        Dim rs As SqlClient.SqlDataReader
        Dim Ssql As String = "SELECT * FROM DirectorioList " & " WHERE codigo='" & CodigoAlex & "' or cedulaidentidadruc ='" & CodigoAlex & "'  " & seleccion & ""
        Dim comm As New SqlClient.SqlCommand(Ssql, ConxAdcom)
        ConxAdcom.Open()
        rs = comm.ExecuteReader

        If rs.Read = False Then rs.Close() : CargarAlex = False : Exit Function

        With rs
            codigo = .Item("codigo").ToString()
            If Not IsDBNull(.Item("nombres")) Then nombres = .Item("nombres").ToString()
            If Not IsDBNull(.Item("apellidos")) Then apellidos = .Item("apellidos").ToString()
            If Not IsDBNull(.Item("NombreImpresion")) Then NombreImpresion = .Item("NombreImpresion").ToString()
            If Not IsDBNull(.Item("CedulaIdentidadRuc")) Then CiRuc = .Item("CedulaIdentidadRuc").ToString()
            If Not IsDBNull(.Item("telefono1")) Then telefono1 = .Item("telefono1").ToString()
            If Not IsDBNull(.Item("Domicilio")) Then direccion = .Item("Domicilio").ToString()
            If Not IsDBNull(.Item("Ciudad")) Then Ciudad = .Item("Ciudad").ToString()
            If Not IsDBNull(.Item("TipoIdentificacion")) Then TipoId = .Item("TipoIdentificacion").ToString()
            If Not IsDBNull(.Item("TipoPersona")) Then TipoPersona = IIf(.Item("TipoPersona").ToString() = "N", "1", "2").ToString()
            If Not IsDBNull(.Item("VendedorInterno")) Then CodVendedor = .Item("VendedorInterno").ToString()
            If Not IsDBNull(.Item("ZonaVentas")) Then ZonaVentas = .Item("ZonaVentas").ToString()
            If Not IsDBNull(.Item("TipoCli")) Then TipoCliente = .Item("TipoCli").ToString()
            NombreCli = CorrijeNulo(.Item("NombreCli"))
        End With
        rs.Close()
        CargarAlex = True
        ConxAdcom.Close()
        ConxAdcom.Dispose()
        rs.Close()
        comm.Dispose()
    End Function
    Public Shared Sub actualizarDirectorio(txtcedula As String, txtCorreoElectronico As String, txtnombrecliente As String, txtApellido As String, txttelefono2 As String, txttelefono As String, txtdireccion As String, txtSector As String)
        Dim insertar = "update identificacion set HistoriaClinica = '" + txtcedula + "'"
        insertar += ", correoElectrónico = '" + txtCorreoElectronico + "'"
        insertar += ", NombreImpresion = '" + txtnombrecliente + "'"
        insertar += ", Telefono1 = '" + txttelefono + "'"
        insertar += ", Telefono2 = '" + txttelefono2 + "'"
        insertar += ", Domicilio = '" + txtdireccion + "'"
        insertar += ", sector = '" + txtSector + "'"
        insertar += " where Codigo = '" + txtcedula + "' or CedulaIdentidadRuc = '" + txtcedula + "' "

        Using conn As New SqlConnection(datosEmpresa.strConxAdcom)
            conn.Open()
            Using comm As New SqlCommand(insertar, conn)
                comm.ExecuteNonQuery()
            End Using
        End Using

    End Sub

End Class

File: MantenimientoDaxDirectorio/MantenimientoDirectorio/Directorio.cs
using System;
using System.Data.SqlClient;
using System.Runtime.InteropServices;
using DattCom;
using Microsoft.VisualBasic;

namespace directMnt
{
	public class Directorio
	{
		// 'Esta Clase sirve para cargar directorios en produccion o sistemas que  permito que no usen el adcomdx
		public string codigo; // copia local
		public string NombreImpresion; // copia local
		public string CiRuc; // copia local
		public string telefono1; // copia local
		public string telefono2; // copia local
		public string direccion; // copia local
		public string CodVendedor; // copia local
		public string ZonaVentas; // copia local
		public string TipoCliente;
		public string Ciudad;
		public string TipoId;
		public string NombreCli;
		public string TipoPersona;
		// variables locales para almacenar los valores de las propiedades

		public bool CargarAlex(ref string CodigoAlex, ref bool ClioPro, [Optional, DefaultParameterValue("")] ref string SoloCodigo)
		{
			bool CargarAlexRet = default;

			string apellidos = "";
			string nombres = "";
			string seleccion = "";

			var ConxAdcom = new SqlConnection(datosEmpresa.strConxAdcom);
			SqlDataReader rs;
			string Ssql = "SELECT * FROM DirectorioList " + " WHERE codigo='" + CodigoAlex + "' or cedulaidentidadruc ='" + CodigoAlex + "'  " + seleccion + "";
			var comm = new SqlCommand(Ssql, ConxAdcom);
			ConxAdcom.Open();
			rs = comm.ExecuteReader();

			if (rs.Read() == false)
			{
				rs.Close();
				CargarAlexRet = false;
				return CargarAlexRet;
			}

			codigo = rs["codigo"].ToString();
			if (!(rs["nombres"] is DBNull))
				nombres = rs["nombres"].ToString();
			if (!(rs["apellidos"] is DBNull))
				apellidos = rs["apellidos"].ToString();
			if (!(rs["NombreImpresion"] is DBNull))
				NombreImpresion = rs["NombreImpresion"].ToString();
			if (!(rs["CedulaIdentidadRuc"] is DBNull))
				CiRuc = rs["CedulaIdentidadRuc"].ToString();
			if (!(rs["telefono1"] is DBNull))
				telefono1 = rs["telefono1"].ToString();
			if (!(rs["Domicilio"] is DBNull))
				direccion = rs["Domicilio"].ToString();
			if (!(rs["Ciudad"] is DBNull))
				Ciudad = rs["Ciudad"].ToString();
			if (!(rs["TipoIdentificacion"] is DBNull))
				TipoId = rs["TipoIdentificacion"].ToString();
			if (!(rs["TipoPersona"] is DBNull))
				TipoPersona = Interaction.IIf(rs["TipoPersona"].ToString() == "N", "1", "2").ToString();
			if (!(rs["VendedorInterno"] is DBNull))
				CodVendedor = rs["VendedorInterno"].ToString();
			if (!(rs["ZonaVentas"] is DBNull))
				ZonaVentas = rs["ZonaVentas"].ToString();
			if (!(rs["TipoCli"] is DBNull))
				TipoCliente = rs["TipoCli"].ToString();
			NombreCli = Valores.CorrijeNulo(rs["NombreCli"]);
			rs.Close();
			CargarAlexRet = true;
			ConxAdcom.Close();
			ConxAdcom.Dispose();
			rs.Close();
			comm.Dispose();
			return CargarAlexRet;
		}
		public static void actualizarDirectorio(string txtcedula, string txtCorreoElectronico, string txtnombrecliente, string txtApellido, string txttelefono2, string txttelefono, string txtdireccion, string txtSector)
		{
			string insertar = "update identificacion set HistoriaClinica = '" + txtcedula + "'";
			insertar += ", correoElectrónico = '" + txtCorreoElectronico + "'";
			insertar += ", NombreImpresion = '" + txtnombrecliente + "'";
			insertar += ", Telefono1 = '" + txttelefono + "'";
			insertar += ", Telefono2 = '" + txttelefono2 + "'";
			insertar += ", Domicilio = '" + txtdireccion + "'";
			insertar += ", sector = '" + txtSector + "'";
			insertar += " where Codigo = '" + txtcedula + "' or CedulaIdentidadRuc = '" + txtcedula + "' ";

			using (var conn = new SqlConnection(datosEmpresa.strConxAdcom))
			{
				conn.Open();
				using (var comm = new SqlCommand(insertar, conn))
				{
					comm.ExecuteNonQuery();
				}
			}

		}

	}
}